app/controllers/application_controller.rb

```diff
@@ -1,2 +1,9 @@
 class ApplicationController < ActionController::Base
+  def after_sign_in_path_for(resource)
+    if current_user.admin?
+      admin_welcome_index_path
+    else
+      new_search_path
```

Comment:
You can use this construction:
 current_user.admin? ? admin_welcome_index_path : new_search_path